Wizardry/Core/BattleSys/Source/BattleHit.c

```diff
@@ -212,15 +211,35 @@ void BattleGenerateHitAttributes(struct BattleUnit * attacker, struct BattleUnit
         else
         {
 #if (defined(SID_Pavise) && (SID_Pavise < MAX_SKILL_NUM))
-            if (CheckBattleSkillActivte(defender, attacker, SID_Pavise, GetUnitSkill(_tmpunit)))
+            if (CheckBattleSkillActivte(defender, attacker, SID_Pavise, GetUnitSkill(unit_def)))
             {
                 RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_Pavise);
                 gBattleStats.damage = 0;
             }
 #endif
         }
     }
+    if (gBattleStats.damage > 0)
+    {
+#if (defined(SID_KeenFighter) && (SID_KeenFighter < MAX_SKILL_NUM))
+        if (SkillTester(unit_def, SID_KeenFighter))
+        {
+            if (CheckCanTwiceAttackOrder(defender,attacker))
+            {
+                RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_KeenFighter);
+                gBattleStats.damage = gBattleStats.damage*3/4;
+            }
+        }
+#endif
 
+#if (defined(SID_DragonSkin) && (SID_DragonSkin < MAX_SKILL_NUM))
+        if (SkillTester(unit_def, SID_DragonSkin))
+        {
+            RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_DragonSkin);
+            gBattleStats.damage = gBattleStats.damage/2;
```

Comment:
code style issue

Reply:
I'm not sure what the issue here is, sorry. Is it `gBattleStats.damage = gBattleStats.damage/2;` ?